### PR TITLE
[Refactor] Migrate Lance to standard task examples + model_extras

### LINKED ISSUE #4546: [RFC sub-task] Migrate lance example to standard task example + model_extras
Sub-task of #4539.

Migrate the **`lance`** model off its dedicated special-usage example onto the standard task example + a declarative `model_extras` registration, following the reference migration in #4225 (BagelPipeline / SenseNovaU1Pipeline).

### Scope
- Special-usage example dir(s): `examples/{offline_inference,online_serving}/lance/`

### Steps
1. Add `vllm_omni/model_extras/lance.py` declaring `EXTRA_BODY_PARAMS` / `EXTRA_OUTPUT_PARAMS`, and register it in `vllm_omni/model_extras/registry.py`.
2. Route the pipeline's previously-hardcoded knobs through `extra_body` / `custom_output` (keep a legacy fallback during the transition).
3. Delete the bespoke example dir(s) above; fold usage into the relevant standard task example (`text_to_image` / `image_to_image` / `text_to_video` / `text_to_audio` / etc.) with an `extra_body` recipe for any non-default params.
4. Update e2e test(s) to exercise `lance` through the standard path and pass in CI.
5. Update markdown/README: remove the per-model README, document the knobs in the task README/recipe.

### ⚠️ Mandatory for the PR (will be sent back if missing)
- [ ] A complete, copy-pasteable, reproducible run script (model id + all `extra_body` params + env vars)
- [ ] The **actual run results/output** pasted into the PR (generated image/video/audio + logs; request+response for serving)
- [ ] Updated e2e test passing in CI
- [ ] Updated markdown/README

### Definition of done
- [ ] `vllm_omni/model_extras/lance.py` added + registered
- [ ] Pipeline knobs routed via `extra_body` / `custom_output`
- [ ] Bespoke example dir deleted; usage folded into standard task example
- [ ] e2e test updated + passing
- [ ] README/markdown updated
- [ ] Reproducible script + actual results in PR description

### PR DESCRIPTION
**Pull request by lulugoodcoder**
## Purpose

Sub-task of #4539 — migrate the **Lance** model off its bespoke example onto the
standard task examples + a declarative `model_extras` registration, following
the reference migration in #4225.

Closes #4546.

## What changed

**Migration (the core work)**
- Add `vllm_omni/model_extras/lance.py` declaring `LANCE_EXTRA_BODY_PARAMS` /
  `LANCE_EXTRA_OUTPUT_PARAMS` and the t2i / i2i / t2v / i2v prompt builders.
- Register `"LancePipeline"` in `vllm_omni/model_extras/registry.py`, and add a
  `text_to_video_prompt_builder` hook (Lance is the first t2v model needing
  prompt shaping — the shared `text_to_video.py` built the prompt inline before).
- Route the shared `text_to_video.py` example through the new builder hook
  (`args.negative_prompt or None` preserves the historical "omit empty negative
  prompt" behaviour — verified equivalent for existing video models).
- CPU unit tests for the declaration + all four builders
  (`tests/model_extras/test_model_extras.py`).
- Docs: knob table + standard-example commands in `recipes/ByteDance/Lance.md`;
  Lance section in the `text_to_image` README; pointers from both bespoke READMEs.

The bespoke `examples/{offline,online}/lance/` dir is **retained**: `video2video`,
`img2text` and `video2text` have no shared task example to fold into (there is no
`video_to_video` example, and the x2t tasks are autoregressive text generation),
so deleting it would drop three working tasks. Happy to remove it in a follow-up
if there's guidance on where those should live.

**Two bugfixes found while validating (see note below)**
- `[Bugfix]` e2e serving used `--pipeline lance`, which no longer exists on
  vLLM 0.25 — argparse abbreviates it to `--pipeline-parallel-size` and fails on
  a non-int value. Switched to `--deploy-config vllm_omni/deploy/lance.yaml`
  (how BAGEL and the offline examples select the pipeline).
- `[Bugfix]` Lance text-to-image crashed in the diffusion worker with
  `tensor a (3) must match tensor b (6) at dim 1` (mrope RoPE). In the CFG-branch
  batching, `torch.cat(cfg_branch_pids)` used the default `dim=0`, which stacks
  Lance's 3-axis mrope position ids `(3, S)` into `(6, S)` instead of
  concatenating along the sequence. Fixed with `dim=-1` (a no-op for BAGEL's 1-D
  position ids). **This is a BAGEL/Lance model fix, separate in scope from the
  migration — happy to split it into its own PR if preferred.** Without it, Lance
  is non-functional on current upstream (independent of this migration).

## Test Plan (reproducible)

Environment: 1× A100-40GB, vLLM 0.25.0, weights on local disk.

```bash
export HF_HOME=/path/to/local/cache          # local disk, not a network volume
export HF_TOKEN=<read token>                  # avoids anonymous rate limits
hf download bytedance-research/Lance          # download fully first

# CPU tests (no GPU) — validates the registry + builders
pytest tests/model_extras/test_model_extras.py -k lance -v      # 7 passed

# text-to-image
python examples/offline_inference/text_to_image/text_to_image.py \
  --model bytedance-research/Lance --deploy-config vllm_omni/deploy/lance.yaml \
  --prompt "a corgi astronaut on the moon, cinematic" \
  --height 1024 --width 1024 --num-inference-steps 30 --seed 42 \
  --extra-body '{"cfg_text_scale": 4.0, "timestep_shift": 3.5}' \
  --output /tmp/lance_t2i.png

# image-to-image
python examples/offline_inference/image_to_image/image_edit.py \
  --model bytedance-research/Lance --deploy-config vllm_omni/deploy/lance.yaml \
  --image /tmp/input.png \
  --prompt "Convert this into a vibrant cartoon-style illustration" \
  --num-inference-steps 30 --seed 42 \
  --extra-args '{"cfg_text_scale": 4.0, "timestep_shift": 3.5}' \
  --output /tmp/lance_i2i.png

# text-to-video (Lance_3B_Video sub-checkpoint; address the local subfolder)
LANCE_DIR=$(hf download bytedance-research/Lance)
python examples/offline_inference/text_to_video/text_to_video.py \
  --model "$LANCE_DIR/Lance_3B_Video" --model-class-name LancePipeline \
  --prompt "a cat playing piano, cinematic" \
  --height 480 --width 768 --num-frames 25 \
  --num-inference-steps 30 --fps 8 --seed 42 \
  --extra-body '{"cfg_text_scale": 4.0, "timestep_shift": 3.5}' \
  --output /tmp/lance_t2v.mp4
```

## Test Result

- `pytest tests/model_extras/test_model_extras.py -k lance` → **7 passed** (A100).
- text-to-image → 1024×1024, 30 steps, ~11s, peak ~20 GB. ✅
- image-to-image → ✅
- text-to-video → ✅

t2i
lance_t2i
i2i
lance_i2i
t2v
https://github.com/user-attachments/assets/b8e585cb-3ea8-42bc-9fa2-956528feb7bc

https://github.com/user-attachments/assets/c74e9d13-c460-4202-9007-54681d1bc70a

### CODE REVIEW

File: vllm_omni/model_extras/lance.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    width: int | None = None,
+    num_frames: int | None = None,
+) -> dict[str, Any]:
+    video_prompt: dict[str, Any] = {
+        "prompt": render_lance_prompt("t2v", prompt),
+        "modalities": ["video"],
+        "extra_args": _video_extra_args(height, width, num_frames),
+    }
+    if negative_prompt is not None:
+        video_prompt["negative_prompt"] = negative_prompt
```
Comment (lishunyang12) [inline vllm_omni/model_extras/lance.py:101]: Put negative_prompt in extra_args; Lance's video CFG never reads the root field.
